lib/proc/proc_snippet.c

```diff
@@ -250,20 +260,49 @@ func_snippet_html(grn_ctx *ctx, int nargs, grn_obj **args,
 {
   grn_obj *snippets = NULL;
 
-  /* TODO: support parameters */
-  if (nargs == 1) {
+  if (nargs > 0) {
     grn_obj *text = args[0];
     grn_obj *expression = NULL;
     grn_obj *condition_ptr = NULL;
     grn_obj *condition = NULL;
     grn_obj *snip = NULL;
+    grn_obj *default_return_value = NULL;
     int flags = GRN_SNIP_SKIP_LEADING_SPACES;
     unsigned int width = 200;
     unsigned int max_n_results = 3;
     const char *open_tag = "<span class=\"keyword\">";
     const char *close_tag = "</span>";
     grn_snip_mapping *mapping = GRN_SNIP_MAPPING_HTML_ESCAPE;
 
+    if (nargs > 1 && args[1]->header.type == GRN_TABLE_HASH_KEY) {
+      grn_obj *options = args[1];
+      void *key;
+      uint32_t key_size;
+      grn_obj *value;
+  
+      GRN_HASH_EACH_BEGIN(ctx, (grn_hash *)options, cursor, id) {
+        grn_raw_string option_name;
+
+        grn_hash_cursor_get_key_value(ctx, cursor,
+                                      &key, &key_size,
+                                      (void **)&value);
+        option_name.value = key;
+        option_name.length = key_size;
+        if (GRN_RAW_STRING_EQUAL_CSTRING(option_name, "default")) {
+          default_return_value = value;
+        } else {
+          GRN_PLUGIN_ERROR(ctx, GRN_INVALID_ARGUMENT,
+                           "snippet_html(): invalid option name: <%.*s>",
+                           key_size, (char *)key);
+          grn_hash_cursor_close(ctx, cursor);
+          break;
+        }
+      } GRN_HASH_EACH_END(ctx, cursor);
+      if (!default_return_value) {
```

Comment:
What is expected behavior of `snippet_html("...", {})`?

Reply:
We expect return null in this case.
Therefore, this `if` section(`if (!default_return_value) {` ) is needless.

I'll fix it.

Reply:
Could you add a test for this case?